### PR TITLE
Atomic fixes

### PR DESCRIPTION
**Pull request by Artemy-Mellanox**

### CODE REVIEW

File: src/uct/ib/mlx5/dv/ib_mlx5dv_md.c [hunk 1]
```diff
@@ -672,6 +673,8 @@ static ucs_status_t uct_ib_mlx5_devx_md_open(struct ibv_device *ibv_device,
             }
         }
 
+        dev->atomic_align = ucs_ilog2(arg_size);
```
Comment (brminich) [inline src/uct/ib/mlx5/dv/ib_mlx5dv_md.c:676]: do you really need to save log?

File: src/uct/ib/mlx5/dv/ib_mlx5dv_md.c [hunk 2]
```diff
@@ -673,7 +673,7 @@ static ucs_status_t uct_ib_mlx5_devx_md_open(struct ibv_device *ibv_device,
             }
         }
 
-        dev->atomic_align = ucs_ilog2(arg_size);
+        dev->atomic_align = UCS_BIT(ucs_ilog2(arg_size));
```
Comment (brminich) [inline src/uct/ib/mlx5/dv/ib_mlx5dv_md.c:676]: maybe can do `ucs_roundup_pow2` instead?

Comment (Artemy-Mellanox) [reply src/uct/ib/mlx5/dv/ib_mlx5dv_md.c:676]: need round down. arg_size is bit field of logs of possible sizes, here need max size.

Comment (brminich) [reply src/uct/ib/mlx5/dv/ib_mlx5dv_md.c:676]: ok we have `ucs_align_down_pow2`)

**Comment by yosefe** [comment]
bot:pipe:retest

**Comment by yosefe** [comment]
bot:pipe:retest

File: src/uct/ib/mlx5/dv/ib_mlx5dv_md.c (hunk 2 above)
Comment (yosefe) [reply src/uct/ib/mlx5/dv/ib_mlx5dv_md.c:676]: pls add ucs_round_down_pow2 to sys/math.h

File: src/uct/ib/mlx5/dv/ib_mlx5dv_md.c [hunk 3]
```diff
@@ -241,7 +241,8 @@ static ucs_status_t uct_ib_mlx5_devx_reg_atomic_key(uct_ib_md_t *ibmd,
 
     reg_length = UCT_IB_MD_MAX_MR_SIZE;
     addr       = (intptr_t)mr->addr & ~(reg_length - 1);
-    length     = mr->length + (intptr_t)mr->addr - addr;
+    length     = ucs_align_up(mr->length + (intptr_t)mr->addr - addr,
```
Comment (yosefe) [inline src/uct/ib/mlx5/dv/ib_mlx5dv_md.c:244]: pls add code comment to why we use md->super.dev.atomic_align

**Comment by Artemy-Mellanox** [comment]
bot:retest

File: src/ucs/sys/math.h [hunk 4]
```diff
@@ -73,6 +73,8 @@ BEGIN_C_DECLS
         pow2; \
     })
 
+#define ucs_rounddown_pow2(_n) (ucs_roundup_pow2(_n) / 2)
```
Comment (yosefe) [inline src/ucs/sys/math.h:76]: it's not correct for powers of 2

**Comment by yosefe** [comment]
bot:pipe:retest

**Comment by yosefe** [comment]
@Artemy-Mellanox pls squash

**Pull request merged**